libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/generator/OpenHouseDataLayoutStrategyGenerator.java

```diff
@@ -193,6 +208,22 @@ private DataCompactionConfig buildDataCompactionConfig(
         .build();
   }
 
+  private double computeFileCountReductionDiscount(
+      Dataset<SnapshotStat> snapshotStats, int lookbackDays) {
+    long snapshotCount =
+        snapshotStats
+            .filter(
+                (FilterFunction<SnapshotStat>)
+                    snapshotStat ->
+                        snapshotStat.getCommittedAt()
+                            >= System.currentTimeMillis() - TimeUnit.DAYS.toMillis(lookbackDays))
+            .count();
+    // if there are new snapshots committed in the past N days,
+    // then assume file count reduction is 0,
+    // because the table could have been overwritten
+    return snapshotCount == 0 ? 0.0 : 1.0;
```

Comment:
Should this be a ratio of 
`snapshotsCreatedDuringLookbackPeriod / totalSnapshotsInTable`
what do you think?